foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/v1_9_0/ArgocdTemplateSyncPolicyMigration.java

```diff
@@ -0,0 +1,118 @@
+package com.boozallen.aissemble.upgrade.migration.v1_9_0;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import com.boozallen.aissemble.upgrade.migration.AbstractAissembleMigration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.boozallen.aissemble.upgrade.util.YamlUtils.indent;
+import static org.technologybrewery.baton.util.FileUtils.readAllFileLines;
+import static org.technologybrewery.baton.util.FileUtils.writeFile;
+
+/**
+ * This migration is responsible for updating the ArgoCD values file to include the syncPolicy in order to enable
+ * ArgoCD automatically sync an application ability. Ultimately, this enables the ArgoCD sync wave feature to function properly.
+ */
+public class ArgocdTemplateSyncPolicyMigration extends AbstractAissembleMigration {
+
+    protected static final Logger logger = LoggerFactory.getLogger(ArgocdTemplateSyncPolicyMigration.class);
+    private static final String SYNC_POLICY = "syncPolicy:";
+    private static final String SPEC = "spec:";
+    private static final String SPEC_INDENTION = indent(1);
+
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = true;
+        if (file != null && file.exists()) {
+            try {
+                List<String> content = readAllFileLines(file);
+                if (isApplication(content)) {
+                    for (String line: content) {
+                        if (line.trim().equals(SYNC_POLICY)) {
+                            shouldExecute = false;
+                            break;
+                        }
+                    }
+                } else {
+                    shouldExecute = false;
+                }
+            } catch (IOException e) {
+                logger.error("Failed to check content of file: {}", file.getPath(), e);
+            }
+        }
+        return shouldExecute;
+    }
+
+    @Override
+    protected boolean performMigration(File file) {
+        try {
+            List<String> content = readAllFileLines(file);
+            List<String> updatedContent = new ArrayList<>();
+            boolean startSpecConfig = false;
+            int counter = 0;
+            int size = content.size();
+            for (String line: content) {
+                counter++;
+                boolean lastItem = counter == size;
+
+                line = line.stripTrailing();
+                if (line.equals(SPEC)) {
+                    startSpecConfig = true;
+                    updatedContent.add(line);
+                    continue;
+                }
+
+                if (startSpecConfig && !line.isEmpty() && !line.startsWith(SPEC_INDENTION)) {
```

Comment:
A: Probably better to use the same indentation detection logic as you wrote for the values file migration